src/model/resources/hydrogen/electrolyzer.jl

```diff
@@ -182,13 +189,21 @@ function electrolyzer!(EP::Model, inputs::Dict, setup::Dict)
     ### Objective Function ###
     # Subtract hydrogen revenue from objective function
     scale_factor = setup["ParameterScale"] == 1 ? 10^6 : 1  # If ParameterScale==1, costs are in millions of $
-    @expression(EP,
-        eHydrogenValue[y in ELECTROLYZERS, t in 1:T],
-        (inputs["omega"][t] * EP[:vUSE][y, t] / hydrogen_mwh_per_tonne(gen[y]) *
-         hydrogen_price_per_tonne(gen[y])/scale_factor))
-    @expression(EP,
-        eTotalHydrogenValueT[t in 1:T],
-        sum(eHydrogenValue[y, t] for y in ELECTROLYZERS))
+    @expression(EP, eHydrogenValue[y in ELECTROLYZERS, t in 1:T],
+        omega[t] * EP[:vUSE][y, t] / hydrogen_mwh_per_tonne(gen[y]) *
+        hydrogen_price_per_tonne(gen[y])/scale_factor)
+    if !isempty(VS_ELEC)
+        @expression(EP, eHydrogenValue_vs[y in VS_ELEC, t in 1:T],
+            omega[t] * EP[:vP_ELEC][y, t] / by_rid(y, :hydrogen_mwh_per_tonne_elec) *
+            by_rid(y, :hydrogen_price_per_tonne_elec)/scale_factor)
+    end
+    @expression(EP, eTotalHydrogenValueT[t in 1:T],
+        if !isempty(VS_ELEC)
+            sum(eHydrogenValue[y, t] for y in ELECTROLYZERS) +
+            sum(eHydrogenValue_vs[y, t] for y in VS_ELEC)
+        else
+            sum(eHydrogenValue[y, t] for y in ELECTROLYZERS)
+        end)
```

Comment:
These lines also can be wrapped using the `add_similar_to_expression!`:
```suggestion
    @expression(EP, eTotalHydrogenValueT[t in 1:T],
        sum(eHydrogenValue[y, t] for y in ELECTROLYZERS))
    # Add the revenue from co-located VRE-STOR electrolyzers
    if !isempty(VS_ELEC)
        expr = @expression(EP, [t in 1:T], sum(eHydrogenValue_vs[y, t] for y in VS_ELEC))
        add_similar_to_expression!(EP[:eTotalHydrogenValueT], expr)
    end
```